### PR TITLE
DATAUP-314: Pre-PR file cleanup and prep

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

ESLint + Prettier on the files that are/will be affected by the changes for the job status updating thing.

Also removed some cruft and fixed some issues identified by ESLint.

I've updated the ESLint config to remove some rules that were made obsolete by Prettier, and added in some extra rules to use more of the exciting ES6+ features.

More comments inline.

# Jira Ticket / Issue #

https://kbase-jira.atlassian.net/browse/DATAUP-314

# Testing Instructions
* Details for how to test the PR: 
- [x] Tests pass in travis and locally 
- [x] Changes available by spinning up a local narrative and seeing that nothing has changed.

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works - N/A
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -13,23 +13,30 @@ rules:
   strict:
     - error
     - function
-  indent:
```
Comment (ialarmedalien) [inline .eslintrc.yml:16]: The rules that have been removed are all covered by Prettier.

File: kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStateList.js [hunk 2]
```diff
@@ -24,72 +18,83 @@ define([
...
+                    class: `${cssBaseClass}__table_body`,
+                }),
+            ]
+        );
     }
 
     // Convert the table to a datatable object to get functionality
-    function renderTable(container, numberOfChildJobs){
-        container.find('table').dataTable({
+    function renderTable($container, numberOfChildJobs) {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStateList.js:71]: `container` represents a jQuery object, so I renamed it `$container`.

File: kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStatusTab.js [hunk 3]
```diff
@@ -1,114 +1,111 @@
...
-    'common/ui',
-    'util/jobLogViewer',
-    './jobStateList'
-], function (
+define(['bluebird', 'kb_common/html', 'common/ui', 'util/jobLogViewer', './jobStateList'], (
     Promise,
     html,
     UI,
-    LogViewer,
+    JobLogViewer,
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStatusTab.js:5]: Renamed to use `JobLogViewer` throughout the codebase (in another file, it is referred to as `JobViewer` 🤦 ).

File: kbase-extension/static/kbase/js/common/jobs.js [hunk 4]
```diff
@@ -1,117 +1,6 @@
...
-    /*
-     * Strip out console commands from text captured from console:
-     * http://search.cpan.org/~jlmorel/Win32-Console-ANSI-1.10/lib/Win32/Console/ANSI.pm
-     *
-     */
-    function consoleToText(consoleText) {
-        return consoleText.replace(/\[([\s\S]*?)m/g, '');
-    }
-
-    function runPython(command) {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobs.js:37]: This is not used or exported, so I got rid of it.

File: kbase-extension/static/kbase/js/common/jobs.js [hunk 5]
```diff
@@ -1,117 +1,6 @@
...
-    'bluebird',
-    'base/js/namespace'
-], function (Promise, Jupyter) {
+define([], () => {
     'use strict';
 
-    /*
-     * Exceptions
-     */
-    function JobError(message, remoteStacktrace) {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobs.js:13]: Not used or exported => 🗑️

File: kbase-extension/static/kbase/js/common/jobs.js [hunk 6]
```diff
@@ -125,15 +14,13 @@ define([
...
     function isValidJobState(jobState) {
-        if (typeof jobState === 'object' && jobState !== null) {
+        if (jobState !== null && typeof jobState === 'object') {
             return jobState.hasOwnProperty('created') && jobState.hasOwnProperty('job_id');
         }
         return false;
     }
 
     return {
-        getLogData: getLogData,
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobs.js:135]: `getLogData` and `deleteJob` are not used anywhere => 🗑️

File: kbase-extension/static/kbase/js/util/jobLogViewer.js [hunk 7]
```diff
@@ -379,13 +379,36 @@ define([
...
+         * lines - list, each is a small object with keys (these are all post-processed after fetching):
+         *      line - string, the line text
+         *      isError - boolean, true if that line denotes an error
+         *      ts - int timestamp
+         *      lineNumber - int, what line this is
+         * firstLine - int, the first line we're tracking (inclusive)
+         * lastLine - int, the last line we're tracking (inclusive)
+         * totalLines - int, the total number of lines available from the server as of the last message.
+         */
+        const model = Props.make({
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/jobLogViewer.js:396]: moved up from the bottom of the file.

File: kbase-extension/static/kbase/js/util/jobLogViewer.js [hunk 8]
```diff
@@ -882,63 +902,61 @@ define([
             fsm.newState({ mode: 'job-not-found' });
         }
 
-        var externalEventListeners = [];
+        function handleJobLogs(message) {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/jobLogViewer.js:905]: extracted out into its own function for clarity

File: kbase-extension/static/kbase/js/util/jobLogViewer.js [hunk 9]
```diff
@@ -882,63 +902,61 @@ define([
             fsm.newState({ mode: 'job-not-found' });
         }
 
-        var externalEventListeners = [];
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/jobLogViewer.js:885]: renamed `externalEventListeners` to `listeners`, declared the variable at the top of the file with the other vars.

File: kbase-extension/static/kbase/js/util/jobStateViewer.js [hunk 10]
```diff
@@ -3,22 +3,16 @@
  * The viewer updates based on changes to the job state and view model.
  * It reports the current job state, runtime, and how long it has been / was queued for.
  */
-define([
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/jobStateViewer.js:6]: Moved this file to `util`, where it lives alongside `jobLogViewer.js`

File: kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeJobStatus.js [hunk 11]
```diff
@@ -46,14 +43,14 @@ define([
     Runtime,
     Semaphore,
     utils,
-    JobViewer,
+    JobLogViewer,
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeJobStatus.js:46]: Rename `JobViewer` => `JobLogViewer`

File: nbextensions/appCell2/widgets/tabs/status/logTab.js [hunk 12]
```diff
@@ -7,22 +7,13 @@ define([
     'common/ui',
     'common/runtime',
     'util/jobLogViewer',
-    './jobStateViewer',
+    'util/jobStateViewer',
```
Comment (ialarmedalien) [inline nbextensions/appCell2/widgets/tabs/status/logTab.js:10]: this module has moved

File: nbextensions/appCell2/widgets/tabs/status/logTab.js [hunk 13]
```diff
@@ -33,155 +24,152 @@ define([
...
                 startDetails({
                     jobId: selectedJobId,
-                    isParentJob: true
+                    isParentJob: true,
                 });
 
                 function startDetails(arg) {
-                    var selectedJobId = arg.jobId ? arg.jobId : model.getItem('exec.jobState.job_id');
-                    config.clickedId = selectedJobId;
+                    const jobId = arg.jobId
```
Comment (ialarmedalien) [inline nbextensions/appCell2/widgets/tabs/status/logTab.js:152]: `selectedJobId` is used in the upper scope ==> rename to `jobId`

File: nbextensions/bulkImportCell/bulkImportCell.js [hunk 14]
```diff
@@ -20,7 +20,7 @@ define([
     'common/cellComponents/tabs/jobStatus/jobStatusTab',
     './bulkImportCellStates',
     'common/cellComponents/tabs/results/resultsTab',
-    './testAppObj.js',
```
Comment (ialarmedalien) [inline nbextensions/bulkImportCell/bulkImportCell.js:23]: ARGH!

File: kbase-extension/static/kbase/js/common/ui.js [hunk 15]
```diff
@@ -8,7 +8,7 @@ define([
     'google-code-prettify/prettify',
     'css!google-code-prettify/prettify.css',
     'bootstrap',
-], function ($, Promise, html, Jupyter, Runtime, PR) {
+], ($, Promise, html, Jupyter, Runtime, PR) => {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/ui.js:11]: this file now has 0 eslint and sonarcloud errors! 🎉

File: nbextensions/bulkImportCell/bulkImportCell.js (hunk 14 above)
Comment (briehl) [reply nbextensions/bulkImportCell/bulkImportCell.js:23]: 😦

**Review by briehl — COMMENTED** [review]
Honestly, this is a lot to effectively go through. What I saw and read from your notes is that it's almost entirely changes from prettier and es-lint, with some minor restructuring and renaming. Any new code here that I should review?

Oh, and thanks for deleting the unused functions!

**Comment by ialarmedalien** [comment]
There isn't any novel content -- it is all eslint + prettier, and a few places where I've renamed something to fix eslint errors or deleted cruft. I've generally commented on those instances to bring them to your attention.

Your comment makes me glad that I didn't just include these formatting / linting changes in a PR where I made content changes!

**Comment by briehl** [comment]
I'm also glad you didn't do that! Anyway, I glanced through the code, and tests pass, so that's enough for me.

**Review by briehl — APPROVED** [review]

**Pull request merged**